src/Maestro/Maestro.Web.Tests/SubscriptionsController20200220Tests.cs

```diff
@@ -0,0 +1,485 @@
+                createdSubscription2.Policy.UpdateFrequency.Should().Be(Api.v2018_07_16.Models.UpdateFrequency.None);
+                createdSubscription2.TargetBranch.Should().Be(defaultBranchName);
+                createdSubscription2.SourceRepository.Should().Be(defaultAzdoSourceRepo);
+                createdSubscription2.TargetRepository.Should().Be(defaultAzdoTargetRepo);
+            }
+
+            // List all (both) subscriptions, spot check that we got both
+            {
+                IActionResult result = data.SubscriptionsController.ListSubscriptions();
+                result.Should().BeAssignableTo<ObjectResult>();
+                var objResult = (ObjectResult) result;
+                objResult.StatusCode.Should().Be((int) HttpStatusCode.OK);
+                List<Subscription> listedSubs = ((IEnumerable<Subscription>) objResult.Value).ToList();
+                listedSubs.Count.Should().Be(2);
+                listedSubs[0].Enabled.Should().Be(true);
+                listedSubs[0].TargetRepository.Should().Be(defaultGitHubTargetRepo);
+                listedSubs[1].Enabled.Should().Be(false);
+                listedSubs[1].TargetRepository.Should().Be(defaultAzdoTargetRepo);
+            }
+            // Use ListSubscriptions() params at least superficially to go down those codepaths
+            {
+                IActionResult result = data.SubscriptionsController.ListSubscriptions(defaultAzdoSourceRepo, defaultAzdoTargetRepo, createdSubscription2.Channel.Id, false);
+                result.Should().BeAssignableTo<ObjectResult>();
+                var objResult = (ObjectResult) result;
+                objResult.StatusCode.Should().Be((int) HttpStatusCode.OK);
+                List<Subscription> listedSubs = ((IEnumerable<Subscription>) objResult.Value).ToList();
+                listedSubs.Count.Should().Be(1);
+                listedSubs[0].Enabled.Should().Be(false);
+                listedSubs[0].TargetRepository.Should().Be(defaultAzdoTargetRepo);
+            }
+            // Directly get one of the subscriptions
+            {
+                IActionResult result = await data.SubscriptionsController.GetSubscription(createdSubscription1.Id);
+                result.Should().BeAssignableTo<ObjectResult>();
+                var objResult = (ObjectResult) result;
+                objResult.StatusCode.Should().Be((int) HttpStatusCode.OK);
+                Subscription theSubscription = (Subscription) objResult.Value;
+                theSubscription.Enabled.Should().Be(true);
+                theSubscription.TargetRepository.Should().Be(defaultGitHubTargetRepo);
+            }
+        }
+
+        [Test]
+        public async Task GetAndListNonexistentSubscriptions()
+        {
+            // No subs added, get a random Guid
+            {
+                IActionResult result = await data.SubscriptionsController.GetSubscription(Guid.NewGuid());
```

Comment:
Random is always scary.  Can we pick a GUID that we _know_ isn't in there? GUID is pretty darn huge, so it's probably not a problem.

Reply:
I'm pretty OK with a test that fails 1 in 2^64 of the time but I can do that.

Reply:
done